api/segments/models.py

```diff
@@ -198,6 +198,38 @@ def deep_clone(self) -> "Segment":
 
         return cloned_segment
 
+    def match_rules_to_segment(self, segment: "Segment") -> None:
+        _rules = self.rules.all()
```

Comment:
I think we should come up with a better way of differentiating 'self' and 'other' segment entities here rather than just the leading underscore.

Reply:
Ok I used the word `self` instead of the leading underscore for this method's logic as well as the `matches_rule` method as well.